### DISCUSSION TITLE
How KeeShare work with sync of references

### DISCUSSION THREAD

**Discussion by LLKoder** [keepassxreboot/keepassxc#10121] [Q&A]
There are 2 password bases.
First database has two groups:  `db_1_g_1` and  `db_1_g_2`.
Second database has one group only: `db_2_g_1`.
Group `db_1_g_2` has one entry.
Group `db_1_g_1` has entry with references to entry in group `db_1_g_2`.
Groups `db_1_g_1` and `db_2_g_1` are synchronized using KeeShare.

What do happien if will be changes in entries of `db_1_g_2` or `db_2_g_1`?

I know that nothing will happen if I change the entry of `db_1_g_2` (this feature is not implemented yet - https://github.com/keepassxreboot/keepassxc/issues/7282).

But what will happien if I will do changes in entry of `db_2_g_1`? Will entry of `db_1_g_2` be overwritten or will only entry of `db_1_g_1` be overwritten?

**Comment by LLKoder** [comment]
As I understand when I create a some entry in `db_1_g_1` group it has unique UUID. This UUID are different with UUID of entry in `db_1_g_2` group. During synchronization KeePassXC catch the same UUID and they will be synchronized. I.e. only entry in `db_1_g_1` group will be synchronized. Entry in `db_1_g_2` group will keep without changes.

Developers, @droidmonkey am I right? Unfortunately manual has a little information about it.

**Comment by droidmonkey** [comment]
You can run this experiment yourself and see. I cannot follow what you are saying above, just do it and see what happens. KeeShare syncs based on entry uuid. If an entry is created or moved to a keeshare enabled group, then it will be synced and will continue to be synced forever. This is one of the major downsides to keeshare I hope to fix at some point.
